strawberry_django/permissions.py

```diff
@@ -48,18 +48,28 @@
 from .utils.query import filter_for_user
 from .utils.typing import UserType
 
-try:
-    # Try to use the smaller/faster cache decorator if available
-    _cache = functools.cache  # type: ignore
-except AttributeError:
-    _cache = functools.lru_cache
-
 if TYPE_CHECKING:
     from strawberry.django.context import StrawberryDjangoContext
 
 
 _M = TypeVar("_M", bound=Model)
 
+_sentinel: Any = object()
+_user_or_anonymous_getter: Optional[Callable[[UserType], UserType]] = _sentinel
+
+
+def _get_user_or_anonymous_getter() -> Optional[Callable[[UserType], UserType]]:
+    global _user_or_anonymous_getter  # noqa: PLW0603
```

Comment:
wouldn't this be nicer with the lru cache?

Reply:
Wasn't sure if that was an overkill, but you are probably right! Going to change it

Reply:
Done!